File: 0055-jump-game/0055-jump-game.js
/**
 * @param {number[]} nums
 * @return {boolean}
 */

//Note: i looked at the Neetcode.io solution first.
// i'm just translating my python (/c++/java/c) code to javascript 
var canJump = function(nums) {
    goal = nums.length - 1; 
    for (i = goal; i >= 0; i--){
        if (i + nums[i] >= goal) {
            goal = i; 
        }
    }
    return (goal == 0); 
    
};

File: 0055-jump-game/0055-jump-game.cpp
// Note: i looked at the neetcode solution first, 
// translating my python code to C++ 
class Solution {
public:
    bool canJump(vector<int>& nums) {
        int goal = nums.size() - 1; 
        for (int i = nums.size() - 1; i >= 0; i--) {
            if (i + nums[i] >= goal) {
                goal = i; 
            }
        }
        return (goal == 0); 
        
    }
};

